packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -64,547 +43,759 @@ type TabsOrientationProps =
+    0,
+    activeIndexProp,
+    onTabSelected,
+  );
+
+  return (
+    <TabsWrapper
+      className={wrapperClassName}
+      focusActivationMode={focusActivationMode}
+      color={color}
+      value={`${activeIndex}`}
+      onValueChange={(value) => setActiveIndex(Number(value))}
+      {...rest}
+    >
+      <TabList className={tabsClassName} ref={forwardedRef}>
+        {labels.map((label, index) => {
+          const tabValue = `${index}`;
+          return React.isValidElement(label) ? (
+            React.cloneElement(label as JSX.Element, {
+              value: tabValue,
+            })
+          ) : (
+            <LegacyTab key={index} value={tabValue} label={label} />
+          );
+        })}
+      </TabList>
+
+      {actions && <TabsActions>{actions}</TabsActions>}
 
       {children && (
-        <Box
-          className={cx('iui-tabs-content', contentClassName)}
-          role='tabpanel'
-        >
+        <TabsPanel value={`${activeIndex}`} className={contentClassName}>
           {children}
-        </Box>
+        </TabsPanel>
       )}
-    </Box>
+    </TabsWrapper>
   );
+}) as PolymorphicForwardRefComponent<'div', TabsLegacyProps>;
+LegacyTabsComponent.displayName = 'Tabs';
+
+// ----------------------------------------------------------------------------
+
+type TabLegacyProps = {
+  /**
+   * The main label shown in the tab.
+   */
+  label?: React.ReactNode;
+  /**
+   * Secondary label shown below the main label.
+   */
+  sublabel?: React.ReactNode;
+  /**
+   * Svg icon shown before the labels.
+   */
+  startIcon?: JSX.Element;
+  /**
+   * Control whether the tab is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Custom content appended to the tab.
+   */
+  children?: React.ReactNode;
+  /**
+   * `value` of the tab.
+   *
+   * Will be set by parent `Tabs` component.
+   */
+  value?: string;
 };
 
+/**
+ * Legacy Tab component.
+ * For full functionality use composition API.
+ *
+ * Individual tab component to be used in the `labels` prop of `Tabs`.
+ * @example
+ * const tabs = [
+ *   <Tab label='Label 1' sublabel='Description 1' />,
+ *   <Tab label='Label 2' startIcon={<SvgPlaceholder />} />,
+ * ];
+ */
+const LegacyTab = React.forwardRef((props, forwardedRef) => {
+  const { label, sublabel, startIcon, children, value, ...rest } = props;
+
+  return (
+    <>
+      <Tab {...rest} value={value as string} ref={forwardedRef}>
+        {startIcon && <TabIcon>{startIcon}</TabIcon>}
+        <TabLabel>{label}</TabLabel>
+        {sublabel && <TabDescription>{sublabel}</TabDescription>}
+        {children}
+      </Tab>
+    </>
+  );
+}) as PolymorphicForwardRefComponent<'button', TabLegacyProps>;
+
+// ----------------------------------------------------------------------------
+// exports
+
+export { LegacyTab as Tab };
+
+/**
+ * Tabs organize and allow navigation between groups of content that are related and at the same level of hierarchy.
```

Comment:
should we explain upfront that `value` is used for connecting tabs with panels?

Reply:
Added a short comment.

Reply:
Suggest rewording for clarity:
> `Tabs.Tab` and `Tabs.Panel` can be associated with each other by passing them the same `value`.

Reply:
Added this rewording